src/components/common/passcode/passcode.modal.create.js

```diff
@@ -34,16 +37,26 @@ class CreatePasscodeModal extends PureComponent {
     this.baseModal.resetModal(flow.title);
   }
 
+  handleNewPasscode = (newPasscode) => {
+    let flowIndex;
+    if (WEAK_PASSCODES.includes(newPasscode)) {
+      flowIndex = STATE_TOO_WEAK;
+    } else {
+      flowIndex = STATE_CONFIRM_PASSCODE;
+      this.tempPasscode = newPasscode;
+    }
+    this.setState({ flowIndex });
+    const { title } = _.find(this.flows, { index: flowIndex });
+    this.baseModal.resetModal(title);
+  }
+
   passcodeOnFill = async (passcode) => {
     const { setPasscode } = this.props;
     const { flowIndex } = this.state;
-    let flow = null;
     switch (flowIndex) {
+      case STATE_TOO_WEAK:
       case STATE_NEW_PASSCODE:
-        this.tempPasscode = passcode;
-        this.setState({ flowIndex: STATE_CONFIRM_PASSCODE });
-        flow = _.find(this.flows, { index: STATE_CONFIRM_PASSCODE });
-        this.baseModal.resetModal(flow.title);
+        this.handleNewPasscode(passcode);
```

Comment:
cant you send the index here as an argument?

Reply:
I'm not sure if I got your point but this index is the current one, on the next step, the index is recalculated based on the new state (too weak or good to continue and ask for confirmation)

Reply:
Yep, but no need to recalculate when you have "flowIndex" and you can send it as a param, or I'm missing sthg?